Kotlin: Rest Client Reactive @ClientExceptionMapper not usable

### Describe the bug

When you try to define a [@ClientExceptionMapper](https://quarkus.io/guides/rest-client-reactive#using-clientexceptionmapper) with Kotlin like this:

```kotlin
@RegisterRestClient(baseUri = "https://swapi.dev/api")
interface SwapiRestClient {
    @GET
    @Path("/people/{id}")
    suspend fun getPersonById(@RestPath id: Int, @RestQuery format: SwapiFormat?): Person

    companion object {
        @JvmStatic
        @ClientExceptionMapper
        fun toException(response: Response): RuntimeException? {
            return when (response.status) {
                400 -> NotFoundException("not found")
                else -> null
            }
        }
    }
}
```

the application will fail on startup with the exception:

```
Caused by: java.lang.IllegalStateException: io.quarkus.rest.client.reactive.ClientExceptionMapper is only supported on static methods of REST Client interfaces that take 'javax.ws.rs.core.Response' as a single parameter and return 'java.lang.RuntimeException'.
	at io.quarkus.rest.client.reactive.deployment.ClientExceptionMapperHandler.generateResponseExceptionMapper(ClientExceptionMapperHandler.java:78)
```

However, the `@JvmStatic` annotation should create a normal static interface method in the bytecode since Kotlin 1.3 (see https://kotlinlang.org/docs/java-to-kotlin-interop.html#static-methods).

I can confirm the rest client interface method is usable from Java with a normal static access: 

```java
public class TestStaticAccess {

	public void testStatic() {
		RuntimeException ex = SwapiRestClient.toException(Response.accepted().build());
		if (ex != null) {
			throw ex;
		}
	}
}
```

### Expected behavior

It should be possible to use a rest client with `@ClientExceptionMapper` with Kotlin as shown above. If possible, it would be great to even check if the method exists on the Companion object so we could avoid using `@JvmStatic` and make the usage more Kotlin idiomatic. 

### Actual behavior

App does not start and shows exception that the annotated method would not be valid.

```
Caused by: java.lang.IllegalStateException: io.quarkus.rest.client.reactive.ClientExceptionMapper is only supported on static methods of REST Client interfaces that take 'javax.ws.rs.core.Response' as a single parameter and return 'java.lang.RuntimeException'.
	at io.quarkus.rest.client.reactive.deployment.ClientExceptionMapperHandler.generateResponseExceptionMapper(ClientExceptionMapperHandler.java:78)
```

### How to Reproduce?

Reproducer:
[2022-08-15-quarkus.zip](https://github.com/quarkusio/quarkus/files/9350320/2022-08-15-quarkus.zip)
 
1. Download and unzip
2. start with `./gradlew quarkusDev`
3. observe error

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

JDK 17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

Quarkus 2.11.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

gradle

### Additional information

_No response_

--- a/extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/ClientExceptionMapperHandler.java
+++ b/extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/ClientExceptionMapperHandler.java
@@ -54,8 +54,11 @@ class ClientExceptionMapperHandler {
         MethodInfo targetMethod = null;
         boolean isValid = false;
         if (instance.target().kind() == AnnotationTarget.Kind.METHOD) {
-            if ((instance.target().asMethod().flags() & Modifier.STATIC) != 0) {
-                targetMethod = instance.target().asMethod();
+            targetMethod = instance.target().asMethod();
+            if (ignoreAnnotation(targetMethod)) {
+                return null;
+            }
+            if ((targetMethod.flags() & Modifier.STATIC) != 0) {
                 String returnTypeClassName = targetMethod.returnType().name().toString();
                 try {
                     boolean returnsRuntimeException = RuntimeException.class.isAssignableFrom(
@@ -112,6 +115,12 @@ class ClientExceptionMapperHandler {
         }
 
         return new Result(restClientInterfaceClassInfo.name().toString(), generatedClassName, priority);
+    }
+
+    private static boolean ignoreAnnotation(MethodInfo methodInfo) {
+        // ignore the annotation if it's placed on a Kotlin companion class
+        // this is not a problem since the Kotlin compiler will also place the annotation the static method interface method
+        return methodInfo.declaringClass().name().toString().contains("$Companion");
     }
 
     static class Result {

--- a/extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/RestClientReactiveProcessor.java
+++ b/extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/RestClientReactiveProcessor.java
@@ -288,6 +288,9 @@ class RestClientReactiveProcessor {
             for (AnnotationInstance instance : index.getAnnotations(CLIENT_EXCEPTION_MAPPER)) {
                 ClientExceptionMapperHandler.Result result = clientExceptionMapperHandler
                         .generateResponseExceptionMapper(instance);
+                if (result == null) {
+                    continue;
+                }
                 if (ifaceToGeneratedMapper.containsKey(result.interfaceName)) {
                     throw new IllegalStateException("Only a single instance of '" + CLIENT_EXCEPTION_MAPPER
                             + "' is allowed per REST Client interface. Offending class is '" + result.interfaceName + "'");

--- a/integration-tests/rest-client-reactive-kotlin-serialization/src/main/kotlin/io/quarkus/it/rest/CountriesClient.kt
+++ b/integration-tests/rest-client-reactive-kotlin-serialization/src/main/kotlin/io/quarkus/it/rest/CountriesClient.kt
@@ -1,11 +1,10 @@
 package io.quarkus.it.rest
 
-import javax.ws.rs.Consumes
-import javax.ws.rs.GET
-import javax.ws.rs.POST
-import javax.ws.rs.Path
-import javax.ws.rs.Produces
+import io.quarkus.rest.client.reactive.ClientExceptionMapper
+import org.jboss.resteasy.reactive.ClientWebApplicationException
+import javax.ws.rs.*
 import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
 
 @Path("")
 interface CountriesClient {
@@ -19,4 +18,19 @@ interface CountriesClient {
     @Path("/countries")
     @Produces(MediaType.APPLICATION_JSON)
     fun countries(): List<Country>
+
+    @GET
+    @Path("/notFoundCountries")
+    @Produces(MediaType.APPLICATION_JSON)
+    fun notFoundCountries(): List<Country>
+    companion object {
+        @JvmStatic
+        @ClientExceptionMapper
+        fun toException(response: Response): RuntimeException? {
+            return when (response.status) {
+                404 -> CountriesNotFoundException()
+                else -> ClientWebApplicationException(response.status)
+            }
+        }
+    }
 }

--- a/integration-tests/rest-client-reactive-kotlin-serialization/src/main/kotlin/io/quarkus/it/rest/CountriesNotFoundException.kt
+++ b/integration-tests/rest-client-reactive-kotlin-serialization/src/main/kotlin/io/quarkus/it/rest/CountriesNotFoundException.kt
@@ -1,0 +1,4 @@
+package io.quarkus.it.rest
+
+class CountriesNotFoundException : RuntimeException() {
+}

--- a/integration-tests/rest-client-reactive-kotlin-serialization/src/main/kotlin/io/quarkus/it/rest/CountriesResource.kt
+++ b/integration-tests/rest-client-reactive-kotlin-serialization/src/main/kotlin/io/quarkus/it/rest/CountriesResource.kt
@@ -34,6 +34,22 @@ class CountriesResource(private val json: Json) {
                 .end(result.capital)
         }
 
+        router.route("/call-not-found-countries").blockingHandler { rc: RoutingContext ->
+            val client = RestClientBuilder.newBuilder()
+                .baseUri(URI.create(rc.body.toString()))
+                .build(CountriesClient::class.java)
+            try {
+                client.notFoundCountries()
+                rc.response()
+                    .setStatusCode(200)
+                    .end("OK")
+            } catch (e: CountriesNotFoundException) {
+                rc.response()
+                    .setStatusCode(204)
+                    .end()
+            }
+        }
+
         router.route("/country").blockingHandler { rc: RoutingContext ->
             val body = json.decodeFromString<Country>(rc.bodyAsString)
             body.capital = "Sthlm"

--- a/integration-tests/rest-client-reactive-kotlin-serialization/src/test/kotlin/io/quarkus/it/rest/client/BasicTest.kt
+++ b/integration-tests/rest-client-reactive-kotlin-serialization/src/test/kotlin/io/quarkus/it/rest/client/BasicTest.kt
@@ -28,4 +28,12 @@ open class BasicTest {
         Assertions.assertThat(response.asString()).isEqualTo("OK")
     }
 
+    @Test
+    fun callNotFoundCountries() {
+        val response = RestAssured.with()
+            .body(country)
+            .post("/call-not-found-countries")
+        Assertions.assertThat(response.statusCode).isEqualTo(204)
+    }
+
 }

